usr/share/rear/build/default/990_verify_rootfs.sh

```diff
@@ -191,10 +194,27 @@ for binary in $( find $ROOTFS_DIR -type f \( -executable -o -name '*.so' -o -nam
     fi
     # Have the whole ldd output only in the log:
     Log "$ldd_output"
-    # Show only the missing libraries to the user to not flood his screen with tons of other ldd output lines:
-    PrintError "$( grep 'not found' <<<"$ldd_output" )"
+    # For each 'not found' shared object (i.e. a shared object that was 'not found' by 'ldd')
+    # check whether or not the shared object may exist nevertheless in the ReaR recovery system
+    # and if yes, we may sufficiently safely assume things are OK in the ReaR recovery system
+    # so we do not report it as missing to the user (for debugging we have all in the log)
+    # cf. https://github.com/rear/rear/issues/3021#issuecomment-2165453757
+    not_found_output="$( grep 'not found' <<<"$ldd_output" )"
+    # not_found_output is a string of multiple lines (separated by \n) that look e.g. like
+    #   libsystemd-shared-255.4-1.fc40.so => not found
+    #   /path/to/library => not found
+    while read not_found_library junk ; do
+        # We prefer a simple 'grep -q' pipe over dealing with find -name versus -path options:
+        if actually_found_library="$( find $ROOTFS_DIR -xdev | grep "$not_found_library" )" ; then
```

Comment:
I just noticed that this lines creates output like this which includes the path to the build dir:

```
/usr/lib64/systemd/libsystemd-core-254.so requires libsystemd-shared-254.so which was not found by 'ldd' but exists as /var/tmp/rear.zV03BYu6tmM1jXe/rootfs/usr/lib64/systemd/libsystemd-shared-254.so
```

I'd suggest changing it to read like this as a solution:

```shell
        if actually_found_library="$( cd $ROOTFS_DIR ; find . -xdev | grep "$not_found_library" )" ; then
```

or something similar

Reply:
Showing the user the path to the build dir is crucial
to indicate that it is all about what there is inside
the ReaR recovery system and not what there is on the
original system.

So actually what is not good in that output line is
that the build dir is missing for what requires
the other library.
I.e. it is not /usr/lib64/systemd/libsystemd-core-254.so
which requires something that is "not found" by 'ldd'
but actually it is
/var/tmp/rear.zV03BYu6tmM1jXe/rootfs/usr/lib64/systemd/libsystemd-core-254.so

The really correct output line should therefore be
```
/var/tmp/rear.zV03BYu6tmM1jXe/rootfs/usr/lib64/systemd/libsystemd-core-254.so requires libsystemd-shared-254.so which was not found by 'ldd' but exists as /var/tmp/rear.zV03BYu6tmM1jXe/rootfs/usr/lib64/systemd/libsystemd-shared-254.so
```
but this output line is very long.

So perhaps a shorter (and still correct) output line could be
```
Within /var/tmp/rear.zV03BYu6tmM1jXe/rootfs usr/lib64/systemd/libsystemd-core-254.so requires libsystemd-shared-254.so which was not found by 'ldd' but exists as usr/lib64/systemd/libsystemd-shared-254.so
```
Still long and rather hard to read and understand.

Reply:
Ahh, good point @jsmeix! Maybe we can mention the build dir only once at the beginning of this section and then use the shorter paths as a compromise between correctness and readability?

Reply:
Yes, this is my plan.

Code changes at several places are needed
because files within ROOTFS_DIR must be
shown to the user without leading slash `/`
to avoid confusion with files on the origiinal system.

Cf. my above
https://github.com/rear/rear/pull/3250#issuecomment-2168077072

So the current output like
```
Testing that the recovery system in /var/tmp/rear.py2WKaUvGlU7Ekg/rootfs contains a usable system
/usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libraries were 'ldd' shows 'not found'
/usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsystemd-shared-255.4-1.fc40.so which was not found by 'ldd' but exists as /var/tmp/rear.py2WKaUvGlU7Ekg/rootfs/usr/lib64/systemd/libsystemd-shared-255.4-1.fc40.so
```
should be (at least a bit) shortened to
something like
```
Testing that the recovery system in /var/tmp/rear.py2WKaUvGlU7Ekg/rootfs contains a usable system
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libraries were 'ldd' shows 'not found'
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsystemd-shared-255.4-1.fc40.so which was not found by 'ldd' but exists as usr/lib64/systemd/libsystemd-shared-255.4-1.fc40.so
```

The second line can appear several times
(one for each `binary` with an `actually_found_library`)
and for each second line the third line
the third line can also appear several times
(one for each `not_found_library` for that `binary`).
Furthermore for each `binary` where 'ldd' shows 'not found'
there could be `actually_missing_libraries`
where the third line looks different and then
an additional LogPrintError is shown at the end.

So the output could look like
```
Testing that the recovery system in /var/tmp/rear.py2WKaUvGlU7Ekg/rootfs contains a usable system
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libraries were 'ldd' shows 'not found'
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsystemd-shared-255.4-1.fc40.so which was not found by 'ldd' but exists as usr/lib64/systemd/libsystemd-shared-255.4-1.fc40.so
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsomething-else.so which was not found by 'ldd' but exists as /path/to/libsomething-elseso
path/to/program requires libraries were 'ldd' shows 'not found'
path/to/program requires libthis.so which was not found by 'ldd' but exists as path/to/libthis.so
path/to/something requires libthat.so which could not be found in the ReaR recovery system
...
"ReaR recovery system in '/var/tmp/rear.py2WKaUvGlU7Ekg/' needs additional libraries, check /var/log/rear/rear-hostname.log for details"
```

Reply:
The info that a library was 'not found' by 'ldd' is shown twice,
in the "second line" and for each "second line"
again in each "third line"
so this info could be left out in the "third lines"
which helps to get those long "third lines" shorter
for example like:
```
Testing that the recovery system in /var/tmp/rear.py2WKaUvGlU7Ekg/rootfs contains a usable system
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libraries were 'ldd' shows 'not found'
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsystemd-shared-255.4-1.fc40.so which exists as usr/lib64/systemd/libsystemd-shared-255.4-1.fc40.so
usr/lib64/systemd/libsystemd-core-255.4-1.fc40.so requires libsomething-else.so which exists as path/to/libsomething-else.so
path/to/program requires libraries were 'ldd' shows 'not found'
path/to/program requires libthis.so which was not found by 'ldd' but exists as path/to/libthis.so
path/to/program requires libthat.so which could not be found in the ReaR recovery system
...
"ReaR recovery system in '/var/tmp/rear.py2WKaUvGlU7Ekg/' needs additional libraries, check /var/log/rear/rear-hostname.log for details"
```
but now those "third lines" do no longer make it clear
that a library which was 'not found' by 'ldd'
exists nevertheless in the recovery system.

Reply:
https://github.com/rear/rear/pull/3308
intends to improve things